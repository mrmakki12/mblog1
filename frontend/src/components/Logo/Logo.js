import React from 'react';
// styles 
import './styles/logo.css';

export const Logo = () => {
    return (
        <div className='logo'>
            <img src='/images/markdown.png' alt='Markdown logo'/>
            <h1>BLOG</h1>
        </div>
    )
}