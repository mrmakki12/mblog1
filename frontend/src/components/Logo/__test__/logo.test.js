// test setup
import React from "react";
// component to test
import { Logo } from '../Logo';
// testing
import { cleanup, render, screen } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import renderer from 'react-test-renderer';

afterEach(cleanup);

// test 1 render without crashing
test('render Logo without crashing', () => {
    render(<Logo />);
});

// test 2 make sure it has img
test('make sure Logo has image', () => {
    render(<Logo />);

    expect(screen.getByRole('img')).toHaveAttribute('src', '/images/markdown.png');
});

// test 3 make sure it has text
test('make sure Logo has text', () => {
    const { getByText } = render(<Logo />);

    expect(getByText(/blog/i)).toBeDefined();
});

// snapshot
test('matches snapshot', () => {
    const tree = renderer.create(<Logo />).toJSON();
    expect(tree).toMatchSnapshot();
});