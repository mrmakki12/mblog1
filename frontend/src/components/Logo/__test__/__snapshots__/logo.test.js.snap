// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="logo"
>
  <img
    alt="Markdown logo"
    src="/images/markdown.png"
  />
  <h1>
    BLOG
  </h1>
</div>
`;
